File: src/main/java/com/cuupa/classificator/services/kb/KnowledgeBase.java
package com.cuupa.classificator.services.kb;

import com.cuupa.classificator.services.kb.semantic.SenderToken;
import com.cuupa.classificator.services.kb.semantic.Topic;

import java.util.ArrayList;
import java.util.List;

class KnowledgeBase {

    private List<Topic> topicList;

    private List<SenderToken> senders;

    void clear() {
        topicList.clear();
        senders.clear();
    }

    List<SenderToken> getSenders() {
        return new ArrayList<>(senders);
    }

    void setSenders(List<SenderToken> senders) {
        this.senders = senders;
    }

    List<Topic> getTopics() {
        return new ArrayList<>(topicList);
    }

    void setTopics(List<Topic> topicList) {
        this.topicList = topicList;
    }
}

File: src/main/java/com/cuupa/classificator/services/kb/KnowledgeBase.kt
package com.cuupa.classificator.services.kb

import com.cuupa.classificator.services.kb.semantic.SenderToken
import com.cuupa.classificator.services.kb.semantic.Topic

class KnowledgeBase {

    var topicList: List<Topic> = listOf()
    var senders: List<SenderToken> = listOf()

    fun clear() {
        topicList = listOf()
        senders = listOf()
    }
}